packages/components/src/Combobox/Combobox.types.ts

```diff
@@ -88,55 +88,97 @@ export interface ComboboxSearchProps {
    * The placeholder for the search input.
    */
   placeholder?: string;
+
   /**
    * The value of the search input
    */
   searchValue: string;
+
   /**
    * The open state of the Combobox listbox.
    */
   open: boolean;
+
   /**
    * Setter for the search input value.
    */
   setSearchValue: Dispatch<SetStateAction<string>>;
 }
 
+export interface ComboboxHeaderProps {
+  /**
+   * The noun to be used in the header label.
+   */
+  readonly subjectNoun?: string;
+
+  /**
+   * The number of selected options.
+   */
+  readonly selectedCount: number;
+
+  /**
+   * The function to call when the clear all button is clicked.
+   */
+  readonly onClearAll: () => void;
+
+  /**
+   * The function to call when the select all button is clicked.
+   */
+  readonly onSelectAll: () => void;
```

Comment:
in the vein of keeping components simple, arguably the job of the header should be
- conditionally show some text
- conditionally show a button
- do a thing when the button is pressed

an optimization we _could_ make here is to simplify the props and move the logic out of this, and into our Combobox. so out there we'd figure out if the action should be `ClearAll` or `SelectAll` and then the `Header` takes in a single action as a prop. same thing for the label logic. then, the `Header` just gets a single `Heading` prop or something like that

since I don't reasonably expect this to be re-used, I don't mind this setup and am 100% not asking for these changes, just something to consider!